### PR TITLE
JDK-8275786: New javadoc option to add script files to generated documentation

### PR DESCRIPTION
**Pull request by hns**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8275786](https://bugs.openjdk.java.net/browse/JDK-8275786): New javadoc option to add script files to generated documentation

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to 3887ed0a4d6ec914b7467e78f323e58966546552

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6173/head:pull/6173` \
`$ git checkout pull/6173`

Update a local copy of the PR: \
`$ git checkout pull/6173` \
`$ git pull https://git.openjdk.java.net/jdk pull/6173/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6173`

View PR using the GUI difftool: \
`$ git pr show -t 6173`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6173.diff">https://git.openjdk.java.net/jdk/pull/6173.diff</a>

</details>

### CODE REVIEW

**Comment by liach** [comment]
looks like we no longer need to allow scripts explicitly in class javadocs in order to use these custom scripts? fantastic it seems.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java [hunk 1]
```diff
@@ -60,6 +60,7 @@
     private boolean showTimestamp;
     private DocPath mainStylesheet;
     private List<DocPath> additionalStylesheets = Collections.emptyList();
+    private List<DocPath> additionalScripts = Collections.emptyList();;
```
Comment (jonathan-gibbons) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java:63]: Suggest moving this down a few lines, after `mainBodyScript` and `scripts`.

Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java:63]: While I realize the maybe-useful similarity between `additionalStylesheets` and `additionalScripts`, the similarity between `scripts` and `additionalScripts` leaves me asking, "why both?". I'm  not sure I would go as far as a systemic rename of `additionalScripts` to `userScripts`, but maybe a comment on the declaration would be helpful to future readers.

**Review by jonathan-gibbons — APPROVED** [review]
OK.

At some point, we should revisit the relationship between strings representing files on the command line, `Path`, and `DocFile`, taking JSR 199 and alternate file managers into account.  Somewhere in there is `Paths.get` which should maybe be cleaned up.  But not here.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java (hunk 1 above)
Comment (hns) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java]: Moved declaration down and added a comment.

**Comment by pavelrappo** [comment]
> looks like we no longer need to allow scripts in comments and add them explicitly in class javadocs in order to use these custom scripts? fantastic it seems.

Look @hns, you have a happy user! :-)

**Pull request closed without merging**